src/ec2/prompter.ts

```diff
@@ -16,6 +19,21 @@ function asQuickpickItem(instance: Ec2Instance): DataQuickPickItem<string> {
     }
 }
 
+function getSelectionFromResponse(response: PromptResult<RegionSubmenuResponse<string>>): Ec2Selection {
+    if (isValidResponse(response)) {
+        return handleEc2ConnectPrompterResponse(response)
+    } else {
+        throw new CancellationError('user')
+    }
+}
+
+export async function promptUserForEc2Selection(): Promise<Ec2Selection> {
+    const prompter = createEc2ConnectPrompter()
+    const response = await prompter.prompt()
+
+    return getSelectionFromResponse(response)
```

Comment:
nit: `getSelectionFromResponse` isn't generic and seems unlikely to be re-used. Could be inlined.